test/integration/test_mail.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+import requests
+from typing import List
+from testcontainers.core.container import DockerContainer
+from autosubmit.notifications.mail_notifier import MailNotifier
+from autosubmit.job.job_common import Status
+from typing import cast
+
+SMTP_PORT = 1025
+API_PORT = 8025
+
+
+@pytest.fixture(scope="module")
+def fake_smtp_server():
+    """Start fake SMTP server container"""
+    with DockerContainer("mailhog/mailhog") \
+            .with_exposed_ports(SMTP_PORT, API_PORT) as container:
+
+        smtp_port = container.get_exposed_port(SMTP_PORT)
+        api_port = container.get_exposed_port(API_PORT)
+        smtp_host = container.get_container_host_ip()
+        api_base = f"http://{smtp_host}:{api_port}"
+
+        for _ in range(10):
+            try:
+                requests.get(f"{api_base}/api/v2/messages")
+                break
+            except requests.ConnectionError:
+                import time
+                time.sleep(1)
+
+        yield {
+            "smtp_host": smtp_host,
+            "smtp_port": smtp_port,
+            "api_base": api_base,
+        }
+        requests.delete(f"{api_base}/api/v2/messages")
+
+
+@pytest.fixture
+def mail_notifier(fake_smtp_server, tmp_path):
+    smtp_host = fake_smtp_server["smtp_host"]
+    smtp_port = fake_smtp_server["smtp_port"]
+
+    def expid_aslog_dir(expid):
+        exp_dir = tmp_path / "aslog" / expid
+        exp_dir.mkdir(parents=True)
+        (exp_dir / "dummy_run.log").write_text("Log entry: simulation started.")
+        return exp_dir
+
+    config = type('Config', (), {
+        'MAIL_FROM': 'notifier@localhost',
+        'SMTP_SERVER': f'{smtp_host}:{smtp_port}',
+        'expid_aslog_dir': staticmethod(expid_aslog_dir),
+    })()
+    return MailNotifier(config)
+
+
+def check_metadata(
+        emails: List[str],
+        expected_subject: str,
+        expid: str,
+        sender: str,
+        recipients: List[str]) -> None:
+    subject = [email["Content"]["Headers"]["Subject"][0] for email in emails]
```

Comment:
I'm getting some type warnings here.

![image](https://github.com/user-attachments/assets/18a31efd-ebc0-4441-beee-19a4d67f3139)

I think that's because `emails` is `list[str]` (can use `list` instead of `List`), but we use `email` as an object or dictionary, accessing things like `email["Content"]["Headers"]["Subject"][0]`.

I think the type is incorrect in the function signature.